tachyon/math/polynomials/univariate/radix2_evaluation_domain.h

```diff
@@ -250,9 +249,10 @@ class Radix2EvaluationDomain : public UnivariateEvaluationDomain<F, MaxDegree>,
   }
 
   constexpr void FFTHelperInPlace(Evals& evals) const {
-    uint32_t log_len = static_cast<uint32_t>(base::bits::Log2Ceiling(
-        static_cast<uint32_t>(evals.evaluations_.size())));
-    this->SwapElements(evals, evals.evaluations_.size() - 1, log_len);
+    size_t num_coeffs = vals.evaluations_.size();
+    uint32_t log_n = this->log_size_of_group_;
+    evals.evaluations_.resize(this->size_, F::Zero());
```

Comment:
I'm not sure if we are optimizing the use of swapElements() if we are filling `F::Zero()`s.

Reply:
Looking at the arguments of `SwapElements()`, the second argument is a value derived from **_the size of the polynomial_**, and the third argument is a value derived from **_the domain size_**. The swap operation is ultimately performed within **_the domain size_**, but it is true that it is optimized because it is operated only as large as **_the size of the polynomial_** given by the second argument. 

If we do not use the second argument as `num_coeffs`, which is the value before `resize()`, a situation where `F:Zero()` is swapped may occur. Also, if we run the `SwapElements()` function without filling it with `F:Zero()`, the swap operation goes outside of size (since **_the size of the polynomial_** <= **_the domain size_**).